network/src/peer/mod.rs

```diff
@@ -78,6 +81,16 @@ enum State {
     ShuttingDown(DisconnectReason),
 }
 
+/// Returns a fully configured length-delimited codec for writing and reading
+/// [`NetworkMessage`]es to a socket.
+pub fn message_codec(max_frame_size: usize) -> LengthDelimitedCodec {
+    LengthDelimitedCodec::builder()
+        .max_frame_length(max_frame_size)
+        .length_field_length(4)
+        .big_endian()
+        .new_codec()
+}
+
```

Comment:
I actually do like putting things in separate functions.  Though, this is another refactor that should be in a separate commit.  

After I scrolled down some, I see you're trying to make use of this function in tests, probably to make a more realistic test.  Let's make it another commit around improving the unit tests.

Reply:
ahh, I wanted to extract this so I could use it in the fuzzer since it needs to serialize NetworkMessage's into wire format